backend/api/Controllers/MissionController.cs

```diff
@@ -384,14 +385,27 @@ public async Task<ActionResult<byte[]>> GetMap([FromRoute] string installationCo
         if (scheduledMissionQuery.AreaName != null)
             area = await _areaService.ReadByInstallationAndName(scheduledMissionQuery.InstallationCode, scheduledMissionQuery.AreaName);
 
-        var scheduledMissionDefinition = new MissionDefinition
+        var source = await _sourceService.CheckForExistingEchoSource(scheduledMissionQuery.EchoMissionId);
+        MissionDefinition? existingMissionDefinition = null;
+        if (source == null)
         {
-            Id = Guid.NewGuid().ToString(),
-            Source = new Source
+            source = new Source
             {
-                SourceId = $"{echoMission.Id}",
+                SourceId = $"{scheduledMissionQuery.EchoMissionId}",
```

Comment:
The returned echoMission is a mock when running the unit tests which is why we don't want to reference it here, but scheduledMissionQuery.EchoMissionId and echoMission.Id should always be the same here regardless.

Reply:
I am not a big fan of altering functionality to fit tests, but agree that here there is no problem, this will be brought up in the discussion related to how we structure our tests

Reply:
I'm looking into making the mock handle this regardless.

Reply:
![image](https://github.com/equinor/flotilla/assets/20266900/f2ed2ed5-ee23-4ae6-94e5-0f0b27f644e9)
This approach lets us avoid this change on line 394, but I'm not sure if it is more elegant.